### PR TITLE
Issue/595 network redirect

### PR DESCRIPTION
**Pull request by wanderingstan**
_? Should these fixes be going to staging? I can fix this PR to do that._

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [X] Map any new environment variables with a default value in the Webpack config

### Description:

Fixing #595 - redirect to our dapp on other domains corresponding to diff eth networks.

### CODE REVIEW

File: src/components/web3-provider.js [hunk 1]
```diff
@@ -441,6 +443,17 @@ class Web3Provider extends Component {
     const isProduction = process.env.NODE_ENV === 'production'
     const networkNotSupported = supportedNetworkId != networkId
 
+    // Redirect if we know a DApp instalation that supports their network.
+    if (currentProvider && networkId && isProduction && networkNotSupported) {
+      const url = new URL(window.location)
+      if (networkId==1) {
```
Comment (micahalcorn) [inline src/components/web3-provider.js]: I've pushed for use of strict equality in the past. Unless those env vars come out as strings, I would favor using it here.

File: src/components/web3-provider.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ const networkNames = {
   999: 'Localhost'
 }
 const supportedNetworkId = process.env.ETH_NETWORK_ID || 1 // Default to mainnet
+const mainnetBaseUrl = process.env.MAINNET_BASEURL || 'https://dapp.originprotocol.com'
+const rinkebyBaseUrl = process.env.RINKEBY_BASEURL || 'https://demo.originprotocol.com'
```
Comment (franckc) [inline src/components/web3-provider.js]: Our rinkeby setup is at https://demo.staging.originprotocol.com

**Comment by wanderingstan** [comment]
If the tests pass I'm good to merge. 

(BTW, talking with John, I'm thinking we should move all default env vars into webpack.config.js -- better to have them all in one place. But not something that has to happen right away.)

**Comment by franckc** [comment]
@wanderingstan yes agreed putting all env vars in webpack.config.js would be good. sounds like a clean up for after release...

**Pull request merged**